plugins/woocommerce/client/legacy/css/woocommerce.scss

```diff
@@ -1699,7 +1699,10 @@ p.demo_store,
  * Buttons
  */
 .woocommerce:where(body:not(.woocommerce-block-theme-has-button-styles)),
-:where(body:not(.woocommerce-block-theme-has-button-styles)) .woocommerce {
+:where(body:not(.woocommerce-block-theme-has-button-styles)):where(
+		:not(.edit-post-visual-editor *)
```

Comment:
I am confused by this change, this removes any styling from the buttons within the editor, you would think we want that?
Do you have a reference to an issue or discussion that suggested we shouldn't do this?

Reply:
This code addresses the second point I mentioned in the PR description: the `Add to cart` button changes its color after the `woocommerce` class is added to the block in the editor.

The scope of those styles wasn’t specific enough, causing some frontend styles to be unintentionally applied to backend blocks in the editor. That’s why I added the `:not(.edit-post-visual-editor *)` to ensure proper scoping.

Reply:
Gotcha, makes sense than, although it also slightly weirds me out. As I thought the site editor should depict the frontend styles, as it is a visual editing experience.

Reply:
> As I thought the site editor should depict the frontend styles, as it is a visual editing experience.

That’s not happening at the moment 😕, but it might be a good idea to explore depicting the frontend styles in the editor. I do wonder if that would affect or potentially break the editing experience.